specs/WebNotification.md

```diff
@@ -0,0 +1,610 @@
+
+# Examples
+
+## Handle PermissionRequested event
+
+This can be achieved with the existing `PermissionRequested` events.
+`PermissionRequested` event used to not be raised for `PermissionKind.Notification`.
+Now, `PermissionRequested` events are raised for `PermissionKind.Notification`, and `PermissionState` needs to be set `Allow` explicitly to allow such permission requests. `PermissionState.Default` for `PermissionKind.Notification` is considered denied.
+
+### C# Sample
+```csharp
+IDictionary<(string, CoreWebView2PermissionKind, bool), bool> _cachedPermissions = 
+    new Dictionary<(string, CoreWebView2PermissionKind, bool), bool>();
+...
+void WebView_PermissionRequested(object sender, CoreWebView2PermissionRequestedEventArgs args)
+{
+    CoreWebView2Deferral deferral = args.GetDeferral();
+    System.Threading.SynchronizationContext.Current.Post((_) =>
+    {
+        using (deferral)
+        {
+            (string, CoreWebView2PermissionKind, bool) cachedKey = (args.Uri, args.PermissionKind, args.IsUserInitiated);
+            if (_cachedPermissions.ContainsKey(cachedKey))
+            {
+                args.State = _cachedPermissions[cachedKey]
+                    ? CoreWebView2PermissionState.Allow
+                    : CoreWebView2PermissionState.Deny;
+            }
+            else
+            {
+                string message = "An iframe has requested device permission for ";
+                message += NameOfPermissionKind(args.PermissionKind) + " to the website at ";
+                message += args.Uri + ".\n\nDo you want to grant permission?\n";
+                message += args.IsUserInitiated ? "This request came from a user gesture." : "This request did not come from a user gesture.";
+                var selection = MessageBox.Show(
+                    message, "Permission Request", MessageBoxButton.YesNoCancel);
+                switch (selection)
+                {
+                    case MessageBoxResult.Yes:
+                        args.State = CoreWebView2PermissionState.Allow;
+                        break;
+                    case MessageBoxResult.No:
+                        args.State = CoreWebView2PermissionState.Deny;
+                        break;
+                    case MessageBoxResult.Cancel:
+                        args.State = CoreWebView2PermissionState.Default;
+                        break;
+                }
+            }
```

Comment:
_cachedPermissions is never added to in this sample code. Is that intentional? Sort of confusing